app/views/admin/secrets/edit.html.erb

```diff
@@ -1,16 +1,47 @@
 <% id = params[:id] %>
 <h1><%= id ? "Edit #{id}" : "New Secret" %></h1>
 
+<div id="deployGroupList" style="display: none;" imbededData="<%= @deployment_group_list.to_json %>" class="hidden"></div>
+
 <section>
   <% url = (id ? admin_secret_path(params[:id]) : admin_secrets_path) %>
   <% method = (id ? :put : :post) %>
-  <% project_permalink, key = id.to_s.split('/', 2) %>
+  <% parse_environment_permalink = id.to_s.split('/', 4).first %>
+  <% parse_project_permalink, key = id.to_s.split('/', 4).second %>
+  <% parse_deploy_group_permalink = id.to_s.split('/', 4).third %>
+  <% key = id.to_s %>
   <%= form_tag url, class: "form-horizontal", method: method do %>
     <fieldset>
+
+      <div class="form-group">
+        <%= label_tag 'secret[environment_permalink]', "Enviorment", class: "col-lg-2 control-label" %>
+        <div class="col-lg-4">
+          <%= select_tag 'secret[environment_permalink]', options_for_select(@environment_permalinks, parse_environment_permalink), class: "form-control", include_blank: true, disabled: !!id, onchange: "populateProjectSelectList()" %>
+          <% if !!id %>
+            <%= hidden_field_tag 'secret[environment_permalink]',  parse_environment_permalink %>
+          <% end %>
+        </div>
+      </div>
+
       <div class="form-group">
         <%= label_tag 'secret[project_permalink]', "Project", class: "col-lg-2 control-label" %>
         <div class="col-lg-4">
-          <%= select_tag 'secret[project_permalink]', options_for_select(@project_permalinks, project_permalink), class: "form-control", disabled: !!id %>
+          <%= select_tag 'secret[project_permalink]', options_for_select(@project_permalinks, parse_project_permalink), class: "form-control", disabled: !!id %>
+          <% if !!id %>
+            <%= hidden_field_tag 'secret[project_permalink]',  parse_project_permalink %>
+          <% end %>
+        </div>
+      </div>
+
+      <div class="form-group">
+        <%= label_tag 'secret[deploy_group_permalink]', "Deploy Group", class: "col-lg-2 control-label" %>
+        <div class="col-lg-4">
+          <% if !!id %>
+            <%= select_tag 'secret[deploy_group_permalink]', options_for_select(@deploy_group_permalinks, parse_deploy_group_permalink), class: "form-control", disabled: !!id %>
+            <%= hidden_field_tag 'secret[deploy_group_permalink]',  parse_deploy_group_permalink %>
```

Comment:
ignore the value during update ... or enable the select ... but security by UI is not good :)